addons/l10n_be_edi/models/account_invoice.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, fields, api, _
+
+import base64
+
+from odoo.tools.xml_utils import check_with_xsd
+
+
+PATH_TO_XSD = 'l10n_be_edi/data/xsd/2.1/maindoc/UBL-Invoice-2.1.xsd'
+
+
+def to_object(item):
+    '''Helpers to convert recursively item to python object to easily access data
+    in the template to render.
+    item[k] = v => item.k = v
+    [dic1, dic2, ...] => [obj1, obj2, ...]
+    
+    :param item: The item to convert.
+    :return: a python object.
+    '''
+    def to_obj(item):
+        if isinstance(item, dict):
+            return type('temp', (), {k: to_obj(v) for k, v in item.iteritems()})
```

Comment:
The entire thing seems singularly misguided, both the odd conversion thingie and the creation of a new class for each dict you find anywhere in the input. Not to mention the possibility of having dict keys which make literally no sense as object fields (e.g. `3`)